File: js/utils/board.js
/** 게임이 실행될 공간 정의 */
class Board{
    /** 초기화 */
    constructor(size=15, is_first=true){
        this.size = size;
        this.board = new Array(size ** 2).fill(0);
        this.is_first = is_first;
        this.is_win = false;
    }

    /** 돌의 수 얻기 
     * @param {number} pieces
    */
    count(pieces){
        return this.board.filter(item => item === pieces).length;
    }

    /** 좌표값으로 보드에 기입 
     * @param {number} y y 좌표
     * @param {number} x x 좌표
     * @param {number} value 기입할 값
    */
    set(y, x, value){
        this.board[y * this.size + x] = value;
    }

    /** 좌표값으로 보드값 얻기 
     * @param {number} y y 좌표
     * @param {number} x x 좌표
     * @return {number}
    */
    get(y, x){
        return this.board[y * this.size + x];
    }

    /** board class를 깊은 복사하는 메소드 */
    deepcopy(){
        let board = new Board(this.size, this.is_first);
        board.board = [...this.board];
        board.is_win = this.is_win;
        return board;
    }
}

File: train/board.py
# 게임 환경
class Board:
    # 초기화
    def __init__(self, size:int = 15, is_first:bool = True):
        self.size = size
        self.board = [0] * size ** 2
        self.is_first = is_first
        self.is_win = False

    # 돌의 수 얻기
    def count(self, pieces:int):
        return self.board.count(pieces)
    
    # 좌표값으로 보드에 기입
    def set(self, y:int, x:int, value:int):
        self.board[y * self.size + x] = value

    # 좌표값으로 보드값 얻기
    def get(self, y:int, x:int):
        return self.board[y * self.size + x]

    # index로 보드에 접근 가능
    def __setitem__(self, index:int, value:int):
        self.board[index] = value

    # index로 보드에 접근 가능한 이터레이터
    def __getitem__(self, index:int):
        return self.board[index]

    # board 끼리의 + 연산시 행동 지정
    def __add__(self, other):
        board = Board(self.size, self.is_first)
        l = []
        for i, j in zip(self.board, other.board):
            if i == 1 and not self.is_first:
                i = 2
            if j == 1 and not other.is_first:
                j = 2
            l.append(i + j)
        board.board = l
        return board
        

    # 형식으로 표시
    def __str__(self):
        string = "    0  1  2  3  4  5  6  7  8  9 10 11 12 13 14\n"
        for i in range(0, 225, 15):
            string += "{0:2} [{1}]\n".format(i//15, ", ".join(map(str, self.board[i:i+15])))
        return string



if __name__ == "__main__":
    b1 = Board()
    b1.set(6, 6, 1)
    b2 = Board(15, False)
    b2.set(5, 5, 1)
    print(b1 + b2)